src/consensus/pbft/libbyz/Replica.cpp

```diff
@@ -536,11 +541,11 @@ void Replica::handle(Request* m)
     {
       if (id() == primary())
       {
-        if (rqueue.append(m))
+        if (rqueue.append(m) && !wait_for_network_to_open)
         {
           send_pre_prepare();
-          return;
         }
+        return;
```

Comment:
If you return here instead of after `send_pre_prepare()` then if `rqueue.append(m)` failes, then we won't reach `delete m` at the end if this function.

Reply:
I still think there is a potential memory leak here. If `rqueue.append(m)` returns false for any reason, and we are waiting for the network to open, then `m` will not be deleted. I think it needs to be:

```
if (rqueue.append(m))
{
  if ( ! wait_for_network_to_open)
  {
    send_pre_prepare();
    return;
  }
}

```